### PR TITLE
pageserver: fix AUX key vectored get validation

### PR DESCRIPTION
**Pull request by VladLazar**
## Problem
The value reconstruct of AUX_FILES_KEY from records is not deterministic since it uses a hash map under the hood.
This caused vectored get validation failures when enabled in staging.

## Summary of changes
Deserialise AUX_FILES_KEY blobs comparing. All other keys should reconstruct deterministically,
so we simply compare the blobs.

## Testing
Tested this locally with data extracted from a tenant which ran into this problem in staging.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pageserver/src/tenant/timeline.rs [hunk 1]
```diff
@@ -911,6 +911,43 @@ impl Timeline {
...
+            // before comparing.
+            let seq_aux_dir_res = AuxFilesDirectory::des(seq);
+            let vec_aux_dir_res = AuxFilesDirectory::des(vec);
+            match (&seq_aux_dir_res, &vec_aux_dir_res) {
+                (Ok(seq_aux_dir), Ok(vec_aux_dir)) => {
+                    assert_eq!(
+                        seq_aux_dir, vec_aux_dir,
+                        "Mismatch for key {} - keyspace={:?} lsn={}: {:?} != {:?}",
+                        key, keyspace, lsn, seq_aux_dir, vec_aux_dir
+                    );
```
Comment (koivunej) [inline pageserver/src/tenant/timeline.rs:935]:
```suggestion
                    assert_eq!(
                        seq_aux_dir, vec_aux_dir,
                        "Mismatch for key {} - keyspace={:?} lsn={}",
                        key, keyspace, lsn, seq_aux_dir, vec_aux_dir
                    );
```

The values are already debug formatted by the macro (and I guess on any problem they'd be large): https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3ccf6c23a34347f1b684e9386610eb09

Or did you find there's some conditionality? I would guess no such thing.

File: pageserver/src/tenant/timeline.rs [hunk 2]
```diff
@@ -911,6 +911,43 @@ impl Timeline {
...
+                }
+                (Err(_), Err(_)) => {}
```
Comment (arpad-m) [inline pageserver/src/tenant/timeline.rs:937]: Maybe log something in this instance? Even if it's just debug level.

**Review by arpad-m — APPROVED** [review]

File: pageserver/src/tenant/timeline.rs (hunk 1 above)
Comment (koivunej) [reply pageserver/src/tenant/timeline.rs:935]: The implementation of `assert_eq!` looks like it does not react to size of left and right.

For the other assert, the `{:?}` of Bytes produces good messages.

**Review by koivunej — APPROVED** [review]
Looking good after the removal of repeated left and right (repeated after assert_eq! macro has already done that).

File: pageserver/src/tenant/timeline.rs (hunk 1 above)
Comment (VladLazar) [reply pageserver/src/tenant/timeline.rs]: Done in https://github.com/neondatabase/neon/pull/7018/commits/a9e0d09e80b01e86267510bae6b98faaf137df6b

File: pageserver/src/tenant/timeline.rs (hunk 2 above)
Comment (VladLazar) [reply pageserver/src/tenant/timeline.rs:937]: I'd rather not. `get` callers already do their own logging on failure to deserialise.

**Pull request merged**